lib/spack/spack/relocate.py

```diff
@@ -351,6 +351,19 @@ def relocate_binary(path_names, old_dir, new_dir, allow_root):
         tty.die("Relocation not implemented for %s" % platform.system())
 
 
+def make_link_relative(cur_path_names, orig_path_names):
+    """
+    Change absolute links to be relative to old_dir
+    """
+    for cur_path, orig_path in zip(cur_path_names, orig_path_names):
+        # We can safely call realpath, all links absolute
```

Comment:
On the one hand I agree that `os.realpath` resolving a chain of symlinks is useful. That being said, if one of them is relative, will the correct path be returned? That doesn't necessarily need to be handled now but a TODO could be worthwhile.

Reply:
That's a spot I missed when I converted things to `os.readlink`, I've changed it now. I agree `os.path.realpath` could be a weakness here.